### ISSUE TITLE
What are the current bugs/issues?

### ISSUE THREAD

**Issue by Ianfeather** [NEURS/gulp-transifex#29]
I'm looking into using this at the moment. You highlight that it is still buggy and needs improvement, do you have a list of these problems? I should hopefully be able to devote some time to it during the process.

Cheers

**Comment by marcoslhc** [comment]
May be we should change the text. Now it's more like a precautioning information for production environment, but works fine even with other formats besides .PO. If you start using it and found anything buggy please open an Issue.

**Comment by Ianfeather** [comment]
Will do. Thanks.

**Comment by Ianfeather** [comment]
Hi @marcoslhc 

I've had zero issues with using this over the last couple of days. It works really nicely and I'd suggest rewording the warning text to be less concerning.

The only decision that I thought was a bit strange was the way translations are downloaded and stored within subdirectories and how the name of the src file has to match the project name. This meant I had to do a little bit of post-processing in order to get files in a workable format for our i18n library. This wasn't a huge deal though.

Thanks for your work on all of this :)

**Comment by marcoslhc** [comment]
Ok. Thanks. Change proposed in https://github.com/NEURS/gulp-transifex/pull/30
I'm closing this for now. But whenever you can post a sample of your post process to see if we can find a workaround together. :beers:

**Comment by marcoslhc** [comment]
please @KevinMartin can you mark this as closed?

**Comment by Ianfeather** [comment]
So my current process is:

I have a language file called `campaign` which I store in a `src` folder. It's named that way because you include the file name when making the request to transifex as part of the api call. Previously (prior to transifex integration) it was just called `en-US.js`.

After fetching from transifex I then rename `campaign` back to `en-US` and put it into the `lang/` folder:

```
return gulp.src('./i18n/campaign')
        .pipe(transifex.pushResource())
        .pipe(transifex.pullResource())
        .pipe($.rename(function (path) {
           path.basename = 'en-US';
           path.extname = '.js';
        }))
        .pipe(gulp.dest('./app/lang'));
```

The second part of the problem is how the files from transifex are saved into folders. So we have:

```
build/
  en_GB/campaign
  sv_SE/campaign
```

but what I actually want is `en-GB.js` and `sv-SE.js`. I presume you put them into folders like this because your projects have multiple translation files in them? Anyway, so I process these files to get them back into `lang/`:

```
return gulp.src('./public/build/transifex/**/*')
        .pipe($.rename(function (path) {
            path.basename = path.dirname.replace(/_/, "-");
            path.extname = '.js';
        }))
       .pipe($.flatten())
       .pipe(gulp.dest('./app/lang'));
```

I hope that's not overly verbose. It isn't too much hassle to bend it into shape but it confused me for a little while.

The other aspect that I found a little strange is that you have to specify a `local_path` when doing `createClient()`. This means that I have to read/write from disk between the original fetch task and the post process one. Not a big deal but doesn't feel like the typical gulp way of dealing with it.

I hope this helps. None of it is an attack on the plugin which as I said has worked really well.
